src/xform_to_datalog/manifest_datalog_facts.cc

```diff
@@ -85,8 +75,8 @@ ManifestDatalogFacts ManifestDatalogFacts::CreateFromManifestProto(
       // Find the ParticleSpec referenced by this Particle. The information
       // contained in the spec will be needed for all facts produced within a
       // Particle.
-      const ir::ParticleSpec &particle_spec =
-          particle_spec_registry.GetParticleSpec(particle_spec_name);
+      const ir::ParticleSpec *particle_spec =
+          CHECK_NOTNULL(system_spec->GetParticleSpec(particle_spec_name));
```

Comment:
If we know this to be non-null, we should use a `const ir::ParticleSpec &` for `particle_spec` to let the type system in on that information.